File: Java/1496.java
class Solution {
    public boolean isPathCrossing(String path) {
        List<List<Integer>> lst = new ArrayList<>();
        int x = 0, y = 0;
        lst.add(Arrays.asList(0,0));
        for(int i = 0; i < path.length(); i++){
            switch(path.charAt(i)){
                case 'N' -> y++;
                case 'S' -> y--;
                case 'E' -> x++;
                default -> x--;
            }
            if(lst.contains(Arrays.asList(x,y))) return true;
            else lst.add(Arrays.asList(x,y));
        }
        return false;
    }
}

File: Python/1496.py
class Solution(object):
    def isPathCrossing(self, path):
        x,y = 0,0
        lst = [[0,0]]
        for i in path:
            if i == "N":
                y += 1
            elif i == "S":
                y -= 1
            elif i == "E":
                x += 1
            else:
                x -= 1
            if [x,y] in lst:
                return True
            else:
                lst.append([x,y])
        return False